### ISSUE TITLE
Signed releases

### ISSUE THREAD

**Issue by aw-was-here** [hadolint/hadolint#245]
It would be great if the binaries on the releases page came with GPG signatures so that we could verify that the downloads haven't been manipulated in bad ways.

**Comment by lorenzo** [comment]
Sounds like a great idea!

Would you like to help doing that? Currently the binaries are uploaded from Travis and Appveyor, so it is a matter of changing the scripts there.

**Comment by zemanlx** [comment]
The process of uploading binaries is automated and transparent (CI logs are public) as @lorenzo already mentioned. If you are worried about an integrity of binaries we can add sha256 hashes for each binary so you can verify later that there were no modifications.

Something like this should work for Linux/OSX
```bash
# Download binary and correnponding sha256sum file
$ wget "https://github.com/hadolint/hadolint/releases/download/v1.10.4/hadolint-Linux-x86_64"
$ wget "https://github.com/hadolint/hadolint/releases/download/v1.10.4/hadolint-Linux-x86_64.sha256"

# Content of file is output of sha256sum program
$ cat hadolint-Linux-x86_64.sha256
5b51f7284d12b78cabee9fb9305c1852bffaba96b4a71db0b68547735aacf902  hadolint-Linux-x86_64

# Check integrity
$ sha256sum -c hadolint-Linux-x86_64.sha256
hadolint-Linux-x86_64: OK
```
and [checksum](https://chocolatey.org/packages/checksum) should be fine for Windows.

@aw-was-here what do you think?

**Comment by aw-was-here** [comment]
If the binaries are compromised, an uploaded sha file will be too.

Signed binaries have the advantage of having a 3rd party acting as an intermediary to guarantee that the bits are what they say they are.  Unfortunately, I have no current experience with travis, so unsure how to securely pass passwords to a process in order to sign them.

**Comment by lorenzo** [comment]
Passwords can be passed as env variables, which will redacted from the output automatically in Travis. We can make sure the password is set. 

@aw-was-here would you like to take care of the rest if we take care of setting the password securely in Travis?

**Comment by zemanlx** [comment]
IMHO if we are in a paranoid mode we should not use CI for this and always sign binaries by hand, presuming that somebody already gets an access to GitHub we cannot trust Travis for this (it is using GitHub auth).
